tests/app/celery/test_letters_pdf_tasks.py

```diff
@@ -232,9 +238,83 @@ def test_get_key_and_size_of_letters_to_be_sent_to_print(notify_api, mocker, sam
     assert len(results) == 3
 
     assert results == [
-        {'Key': '2020-02-16/NOTIFY.REF2.D.2.C.C.20200215180000.PDF', 'Size': 2},
-        {'Key': '2020-02-17/NOTIFY.REF1.D.2.C.C.20200217150000.PDF', 'Size': 1},
-        {'Key': '2020-02-17/NOTIFY.REF0.D.2.C.C.20200217160000.PDF', 'Size': 3},
+        {
+            'Key': '2020-02-16/NOTIFY.REF2.D.2.C.C.20200215180000.PDF',
+            'Size': 2,
+            'ServiceId': str(sample_letter_template.service_id)
+        },
+        {
+            'Key': '2020-02-17/NOTIFY.REF1.D.2.C.C.20200217150000.PDF',
+            'Size': 1,
+            'ServiceId': str(sample_letter_template.service_id)
+        },
+        {
+            'Key': '2020-02-17/NOTIFY.REF0.D.2.C.C.20200217160000.PDF',
+            'Size': 3,
+            'ServiceId': str(sample_letter_template.service_id)
+        },
+    ]
+
+
+@freeze_time('2020-02-17 18:00:00')
+def test_get_key_and_size_of_letters_to_be_sent_to_print_mark_insolvency_letters(
+    notify_api, notify_db_session, mocker
+):
+    # random service
+    service_1 = create_service(service_name="service 1", service_id='f2fe37b0-1301-11eb-aba9-4c3275916899')
+    letter_template_1 = create_template(service_1, template_type=LETTER_TYPE)
+    create_notification(
+        template=letter_template_1,
+        status='created',
+        reference='ref0',
+        created_at=(datetime.now() - timedelta(hours=2))
+    )
+    # insolvency service
+    insolvency_org = create_organisation(organisation_id='f33fdfdd-7533-40cb-b5e8-cd78a1f5d21e', name="Insolvency")
+    insolvency_service = create_service(
+        service_name="insolvency service",
+        service_id='3a5cea08-29fd-4bb9-b582-8dedd928b149',
+        organisation=insolvency_org
+    )
+    insolvency_letter_template = create_template(insolvency_service, template_type=LETTER_TYPE)
+    create_notification(
+        template=insolvency_letter_template,
+        status='created',
+        reference='insolvency',
+        created_at=(datetime.now() - timedelta(hours=3))
+    )
+
+    mock_s3 = mocker.patch('app.celery.tasks.s3.head_s3_object', side_effect=[
+        {'ContentLength': 1},
+        {'ContentLength': 1},
+    ])
+
+    results = get_key_and_size_of_letters_to_be_sent_to_print(datetime.now() - timedelta(minutes=30), postage='second')
+
+    assert mock_s3.call_count == 2
+    mock_s3.assert_has_calls(
+        [
+            call(
+                current_app.config['LETTERS_PDF_BUCKET_NAME'],
+                '2020-02-17/NOTIFY.INSOLVENCY.D.2.C.C.20200217150000.PDF'
```

Comment:
Just a quick cursory glance before stand up, sorry I don't have the time to do a proper review however one thing I think we need to do is keep the filenames of the letters the same and not change those. I don't think we be sure DVLA can support a change in filename and we want to keep this change as small as possible to minimise any portential unexpected side affects. It's definitely OK to change the name on the zip file but I think we need to keep the files inside the same please

Reply:
@idavidmcdonald this is just because pea set the dvla ref to `insolvency` in the test setup on line 283

Reply:
Yeah, the pdf filename did not change, it's just I set ref to `INSOLVENCY`, maybe I could change it to `1N50LV3NCY` so it's less confusing? :D

Reply:
I think don't worry about it for the moment, but thanks. Sorry!